File: C(++)/[2 sem] 2DWorldSimulator/code/src/Commentator.cpp
//
// Created by szymo on 4/24/2023.
//

#include ".headers/Commentator.h"
#include "iostream"

#define MAX_COMMENTS_PER_ROUND 8

void Commentator::comment(const string& comment) {
    comments.push_back(comment);
    currentRoundComments++;
}

void Commentator::newRound() {
    comments.clear();
    currentRoundComments = 0;
    currentRound++;
    comments.push_back("Round " + to_string(currentRound));
}

void Commentator::introduceAuthor() {
    cout << "Author: " << author << endl;
}

void Commentator::commentFight(Organism *attacker, Organism *defender, bool won) {
    string comment = attacker->getSpecieName() + " attacked " + defender->getSpecieName() + " and ";
    if (won) {
        comment += "won";
    } else {
        comment += "lost";
    }
    this->comment(comment);
}

void Commentator::printComments() {
    int counter = 0;
    for (auto &comment : comments) {
        cout << comment << endl;
        counter++;
        if (counter > MAX_COMMENTS_PER_ROUND) return;
    }
}

void Commentator::commentEat(Organism *eater, Organism *eaten) {
    string comment = eater->getSpecieName() + " ate " + eaten->getSpecieName();
    this->comment(comment);
}

int Commentator::getCurrentRound() const {
    return currentRound;
}

void Commentator::setCurrentRound(int newCurrentRound) {
    this->currentRound = newCurrentRound;
    comments.clear();
    comments.push_back("Round " + to_string(currentRound));
}

File: Java/[2 sem] 2DWorldSimulator/code/Commentator.java
import java.util.ArrayList;
import java.util.List;

public class Commentator {
    public final ArrayList<String> comments;
    private int currentRound;
    public int currentRoundComments;
    private final String author;
    private static final int MAX_COMMENTS_PER_ROUND = 20;

    public Commentator(String author) {
        this.comments = new ArrayList<>();
        this.currentRound = 0;
        this.currentRoundComments = 0;
        this.author = author;
    }

    public void comment(String comment) {
        if (currentRoundComments >= MAX_COMMENTS_PER_ROUND) {return;}
        comments.add(comment);
        currentRoundComments++;
    }

    public void newRound() {
        comments.clear();
        currentRoundComments = 0;
        currentRound++;
        comments.add("Round " + currentRound);
    }

    public void commentFight(Organism attacker, Organism defender, boolean won) {
        String comment = attacker.getSpecieName() + " attacked " + defender.getSpecieName() + " and ";
        if (won) {
            comment += "won";
        } else {
            comment += "lost";
        }
        this.comment(comment);
    }

    public void commentEat(Organism eater, Organism eaten) {
        String comment = eater.getSpecieName() + " ate " + eaten.getSpecieName();
        this.comment(comment);
    }

    public int getCurrentRound() {
        return currentRound;
    }

    public void setCurrentRound(int newCurrentRound) {
        currentRound = newCurrentRound;
        comments.clear();
        comments.add("Round " + currentRound);
        currentRoundComments = 1;
    }

    public ArrayList<String> getComments() {
        return comments;
    }
}
